examples/jsm/nodes/accessors/TextureNode.js

```diff
@@ -10,7 +10,7 @@ import { NodeUpdateType } from '../core/constants.js';
 
 class TextureNode extends UniformNode {
 
-	constructor( value, uvNode = null, levelNode = null ) {
+	constructor( value, uvNode = null, levelNode = null, _nodeType = null ) {
```

Comment:
What would be the purpose of `_nodeType` here?

Reply:
It allows to read the texture as its correct format. Currently everything is always overrode to float because it seems that uint textures are used only for envmap so far in the WebGPURenderer which are originally float packed as uint (and so used back as float in shaders).

For example, in the `getIndirectIndex` of the BatchNode function :
https://github.com/mrdoob/three.js/pull/28753/files#diff-2f6dd734ecf239b91329c408e3cc61fd5165c1a6aeac98fb0bf6eb3d78749a42R45-R48

Here the texture is read as usampler2D and returns directly the proper uint format:
`textureLoad( this.batchMesh._indirectTexture, ivec2( x, y ), null, 'uvec4' ).x`


For reference this was the GLSL implementation in the WebGLRenderer:
```glsl
uniform highp usampler2D batchingIdTexture;


float getIndirectIndex( const in int i ) {

int size = textureSize( batchingIdTexture, 0 ).x;
int x = i % size;
int y = i / size;
return float( texelFetch( batchingIdTexture, ivec2( x, y ), 0 ).r );

}
```

This was not possible to make without `_nodeType` as `TextureNode` always overriding the `nodeType` to `vec4<float>` previously.

/cc @sunag

Reply:
I think we could detect the type of texture in `Texture.getNodeType()` if integer or float automatically in `this.value`.

Reply:
This is what I initially did but it breaks all the envmap related example.
So even if we handle the integer/float automatically we will still need to override all the types using a similar logic as `._nodeType` for all the envmap because they are float packed as uint DataTexture.

Reply:
For reference that was this commit:
https://github.com/mrdoob/three.js/pull/28753/commits/c06f0003d4431215d227b2e6d9d17d418cd5ca1f

Reply:
Do you tried something like this?

```js
getNodeType( /*builder*/ ) {

	if ( this.value.isDepthTexture === true ) return 'float';

	if ( this.value.type === UnsignedIntType ) {

		return 'uvec4';

	} else if ( this.value.type === IntType ) {

		return 'ivec4';

	}

	return 'vec4';

}
```

Reply:
It feels more clean indeed! Also it would still be the same issue as it would break envmaps.

Reply:
Did you test? It didn't present the same problem for me.

![image](https://github.com/mrdoob/three.js/assets/502810/45b5b87c-2741-4dca-b486-4ec2e9761e50)

Reply:
Didn't check but this probably breaks in GLSL.

Reply:
So this would be another problem, and not the same one related about envmap. I suggest we follow this path and fix the new issue in GLSLNodeBuilder.

Reply:
By aligning all the logic with the `texture.type` (`UnsignedIntType`, `IntType`) as you demonstrated instead of name of the array buffer constructor everything now works fine without the need of `_nodeType`.

I cleaned up the code consequently. It's now stable on all examples on both backend 👍 

Thanks again for the great review!